docs/apache-airflow/authoring-and-scheduling/deferring.rst

```diff
@@ -96,6 +96,68 @@ When writing a deferrable operators these are the main points to consider:
             # We have no more work to do here. Mark as complete.
             return
 
+
+Writing Triggers
+~~~~~~~~~~~~~~~~
+
+A *Trigger* is written as a class that inherits from ``BaseTrigger``, and implements three methods:
+
+* ``__init__``: A method to receive arguments from operators instantiating it.
+* ``run``: An asynchronous method that runs its logic and yields one or more ``TriggerEvent`` instances as an asynchronous generator.
+* ``serialize``: Returns the information needed to re-construct this trigger, as a tuple of the classpath, and keyword arguments to pass to ``__init__``.
```

Comment:
Maybe it’s better to just put the example code block very early in the sections, and adding docstrings to functions describing each function?

This would be much easier to read than a bunch of paragraphs trying to describe what to do, and only have an example in the end (that can’t be fully understood unless you’ve first read the text)

Reply:
I tried to move the example DAGs upper for these few.